eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/CredentialInfo.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.implementation.util.ImplUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Locale;
+
+public final class CredentialInfo {
+
+    private static final String ENDPOINT = "Endpoint=";
+    private static final String SHARED_ACCESS_KEY_NAME = "SharedAccessKeyName=";
+    private static final String SHARED_ACCESS_KEY = "SharedAccessKey=";
+    private static final String ENTITY_PATH = "EntityPath=";
+
+    private static URI endpoint;
+    private static String sharedAccessKeyName;
+    private static String sharedAccessKey;
+    private static String eventHubName;
+
+    private CredentialInfo() { }
+
+    /**
+     * Create a CredentialInfo object from connection string.
+     *
+     * @param connectionString Connection String
+     * @return CredentialInfo
+     * @throws IllegalArgumentException when 'connectionString' is {@code null} or empty, or cannot be translated into an
+     * {@link CredentialInfo}, or have invalid format.
+     */
+    public static CredentialInfo from(String connectionString) {
+        CredentialInfo credentialInfo = new CredentialInfo();
+        if (ImplUtils.isNullOrEmpty(connectionString)) {
+            throw new IllegalArgumentException(connectionString);
+        }
+
+        String[] args = connectionString.split(";");
+        if (args.length < 3) {
+            throw new IllegalArgumentException("invalid connection string segment count");
```

Comment:
What are we expecting? Should be outlined in the method docs what components there are.

Reply:
Use the error message from: https://github.com/Azure/azure-sdk-for-java/pull/3791/files#diff-a814032d30ff1457abeea40ac2ea360aR64